src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,139 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value, null,tail);
+        if (size == 0) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndexToAdd(index);
+        Node<T> prevNode = getNodeByIndex(index);
+        Node<T> newNode = new Node<>(value, prevNode, prevNode.prev);
+        if (index == 0) {
+            head = newNode;
+        } else {
+            prevNode.prev.next = newNode;
+        }
+        prevNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T listElement : list) {
+            add(listElement);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidator(index);
+        return getNodeByIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexValidator(index);
+        Node<T> node = getNodeByIndex(index);
+        T oldNode = node.element;
+        node.element = value;
+        return oldNode;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexValidator(index);
+        Node<T> removedElement = getNodeByIndex(index);
+        unlink(removedElement);
+        return removedElement.element;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> remove = head;
+        int rem = 0;
+        do {
+            if (object == remove.element
+                    || remove.element != null && remove.element.equals(object)) {
+                remove(rem);
+                return true;
+            }
+            rem++;
+            remove = remove.next;
+        } while (remove.next != null);
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return head == null;
+    }
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T element, Node<T> next, Node<T> prev) {
+            this.element = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
+    private void indexValidator(int index) {
```

Comment:
bad method naming. It's a good practice to make method names verbs 
```suggestion
    private void checkIndex(int index) {
```
OR
```suggestion
    private void validateIndex(int index) {
```